### PR TITLE
NMS-12374: Add tracing for provisiond

### PR DESCRIPTION
**Pull request by cgorantla**
### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-12374
* Bamboo (Continuous Integration): https://bamboo.opennms.org/

### CODE REVIEW

File: opennms-provision/opennms-provisiond/src/main/java/org/opennms/netmgt/provision/service/DefaultProvisionService.java [hunk 1]
```diff
@@ -1453,4 +1470,22 @@ public void setServiceTypeDao(final ServiceTypeDao dao) {
     public void setEventForwarder(final EventForwarder eventForwarder) {
         m_eventForwarder = eventForwarder;
     }
+
+    public Span buildAndStartSpan(String name, SpanContext spanContext) {
+        if(getTracer() == null) {
```
Comment (j-white) [inline opennms-provision/opennms-provisiond/src/main/java/org/opennms/netmgt/provision/service/DefaultProvisionService.java:1475]: Seems odd to mix calls to `getTracer()` and direct access to `m_tracer` in this same method.

File: opennms-provision/opennms-provisiond/src/main/java/org/opennms/netmgt/provision/service/Provisioner.java [hunk 2]
```diff
@@ -214,6 +222,10 @@ public void setMonitoringSystemDao(MonitoringSystemDao monitoringSystemDao) {
         this.monitoringSystemDao = monitoringSystemDao;
     }
 
+    public void setTracerRegistry(TracerRegistry tracerRegistry) {
+        this.m_tracerRegistry = tracerRegistry;
```
Comment (j-white) [inline opennms-provision/opennms-provisiond/src/main/java/org/opennms/netmgt/provision/service/Provisioner.java:226]: `this.m_`

File: opennms-provision/opennms-provisiond/src/test/java/org/opennms/netmgt/provision/service/InvalidRequisitionDataIT.java [hunk 3]
```diff
@@ -143,7 +143,7 @@ public void testImportInvalidAsset() throws Exception {
         // Change it to "pollerCategory" (capital 'C') and the test passes...
         m_provisioner.doImport(invalidAssetFieldResource.getURL().toString(), Boolean.TRUE.toString());
         waitForEverything();
-        m_eventManager.getEventAnticipator().verifyAnticipated();
+            m_eventManager.getEventAnticipator().verifyAnticipated();
```
Comment (j-white) [inline opennms-provision/opennms-provisiond/src/test/java/org/opennms/netmgt/provision/service/InvalidRequisitionDataIT.java:146]: Unnecessary change.

File: opennms-provision/opennms-provisiond/src/test/java/org/opennms/netmgt/provision/service/ProvisionerIT.java [hunk 4]
```diff
@@ -1300,6 +1301,15 @@ public void testProvisionServiceGetScheduleForNodesUponDelete() throws Exception
 
         assertEquals(9, schedulesForNode.size());
         assertEquals(9, m_provisioner.getScheduleLength());
+
+        // Check that re-scheduling won't trigger duplicate scheduling for a given node.
```
Comment (j-white) [inline opennms-provision/opennms-provisiond/src/test/java/org/opennms/netmgt/provision/service/ProvisionerIT.java:1305]: I'm guessing this is related to NMS-12503? :)

Comment (cgorantla) [reply opennms-provision/opennms-provisiond/src/test/java/org/opennms/netmgt/provision/service/ProvisionerIT.java:1305]: yeah, similar issue but https://issues.opennms.org/browse/NMS-12504

File: opennms-provision/opennms-detectorclient-rpc/src/main/java/org/opennms/netmgt/provision/detector/client/rpc/DetectorClientRpcModule.java [hunk 5]
```diff
@@ -86,6 +86,7 @@ public DetectorClientRpcModule() {
                 @Override
                 public DetectorResponseDTO get() {
                     try {
+                        detectRequest.preDetect();
```
Comment (j-white) [inline opennms-provision/opennms-detectorclient-rpc/src/main/java/org/opennms/netmgt/provision/detector/client/rpc/DetectorClientRpcModule.java:89]: We'll need to re-think this for Minion.

**Comment by opennms-bamboo** [comment]
Welcome CI/CD System (cicd-system@opennms.com).

We would like to thank you for your contribution.

I could not find a signed OpenNMS Contributor Agreement (OCA) assigned to the username CI/CD System (cicd-system@opennms.com).
In order to maintain some order with respect to the copyright to the code,
every contributor is asked to sign a "Contributor Agreement".

The OpenNMS Contributor Agreement can be downloaded [here](https://wiki.opennms.org/wiki/images/6/6b/ContributorAgreement.pdf).
Please send a signed copy to oca@opennms.org.

If you need more information, please visit https://wiki.opennms.org/wiki/Contributor_Agreement
